packages/fxa-auth-server/scripts/convert-customers-to-stripe-automatic-tax.ts

```diff
@@ -0,0 +1,328 @@
+  return !subscription.automatic_tax.enabled;
+};
+
+// Checks whether subscription renews too soon to be upgraded to
+// Stripe Automatic Tax
+// 1 month plans require at least 14 days of notice
+// 6 month and yearly plans require at least 30 days of notice
+// This function does not support plans with intervals of less than 1 month
+const isWithinNoticePeriod = (sub: Stripe.Subscription) => {
+  const noSoonerThan = new Date();
+  if (sub.items.data[0].plan.interval === 'month') {
+    noSoonerThan.setUTCDate(noSoonerThan.getUTCDate() + MONTHLY_NOTICE_DAYS);
+  } else {
+    noSoonerThan.setUTCDate(noSoonerThan.getUTCDate() + YEARLY_NOTICE_DAYS);
+  }
+
+  const renewalDate = new Date(
+    sub.current_period_end * SECONDS_TO_MILLISECONDS
+  );
+
+  return noSoonerThan < renewalDate;
+};
+
+// Function returns true if subscription is continuing and
+// will be billed in the future
+const willBeRenewed = (sub: Stripe.Subscription) => {
+  return !sub.cancel_at && !sub.cancel_at_period_end && sub.status === 'active';
+};
+
+// Updates a customer's IP address and enables Stripe automatic tax
+const enableStripeTax = (
+  stripe: Stripe,
+  customerId: string,
+  ipAddressMap: IpAddressMap
+) => {
+  // TODO FXA-6581: Handle ip address map unknown customer
+  return stripe.customers.update(customerId, {
+    tax: {
+      ip_address: ipAddressMap[customerId],
+    },
+  });
+};
+
+// Retrieves a customer record from Stripe
+// Returns null if user is deleted
+const fetchCustomer = async (stripe: Stripe, customerId: string) => {
+  const customer = await stripe.customers.retrieve(customerId, {
+    expand: ['tax'],
+  });
+
+  if (customer.deleted) return null;
+
+  return customer;
+};
+
+// Checks if Stripe customer is in a taxable location
+const isTaxEligible = (customer: Stripe.Customer) => {
+  return customer.tax?.automatic_tax === 'supported';
+};
+
+// Gets a tax amount from a list of tax amounts given it's display name
+// as represented by Stripe
+// This must be done via display name since the tax rate ID changes
+// every time for automatic tax
+const findTaxAmount = (
+  taxAmounts: Stripe.Invoice.TotalTaxAmount[],
+  name: string
+) => {
+  const taxAmount = taxAmounts.find(
+    (taxAmount) => (taxAmount.tax_rate as Stripe.TaxRate).display_name === name
+  );
+
+  return taxAmount?.amount;
+};
+
+// Creates an ordered array of fields destined for CSV format
+// in the order described by the Stripe Tax for Existing Customers PRD
+const buildReport = ({
+  customer,
+  product,
+  plan,
+  subscription,
+  invoicePreview,
+}: {
+  customer: Stripe.Customer;
+  product: Stripe.Product;
+  plan: Stripe.Plan;
+  subscription: Stripe.Subscription;
+  invoicePreview: Stripe.Invoice;
+}) => {
+  const hst = findTaxAmount(invoicePreview.total_tax_amounts, 'HST');
+  const gst = findTaxAmount(invoicePreview.total_tax_amounts, 'GST');
+  const pst = findTaxAmount(invoicePreview.total_tax_amounts, 'PST');
```

Comment:
This looks... Canadian 🇨🇦

Per https://stripe.com/docs/tax/supported-use-cases#canada:-gst,-hst,-pst,-qst,-and-rst, do we need to also potentially account for "QST" (Quebec thing), and/or "RST" (Retail Sales Tax, which is apparently [a Manitoba thing](https://www.gov.mb.ca/finance/taxation/taxes/retail.html)<sup>(tm)</sup>.)

Reply:
Good catch! Will add.